packaging/installer/kickstart-ng.sh

```diff
@@ -0,0 +1,842 @@
+      ;;
+    *)
+      fatal "Unsupported system type detected. Netdata cannot be installed on this system using this script."
+      ;;
+  esac
+}
+
+str_in_list() {
+  printf "%s\n" "${2}" | tr ' ' "\n" | grep -qE "^${1}\$"
+  return $?
+}
+
+confirm_root_support() {
+  if [ "$(id -u)" -ne "0" ]; then
+    if [ -z "${ROOTCMD}" ] && command -v sudo > /dev/null; then
+      ROOTCMD="sudo"
+    fi
+
+    if [ -z "${ROOTCMD}" ] && command -v doas > /dev/null; then
+      ROOTCMD="doas"
+    fi
+
+    if [ -z "${ROOTCMD}" ] && command -v pkexec > /dev/null; then
+      ROOTCMD="pkexec"
+    fi
+
+    if [ -z "${ROOTCMD}" ]; then
+      fatal "We need root privileges to continue, but cannot find a way to gain them. Either re-run this script as root, or set \$ROOTCMD to a command that can be used to gain root privileges"
+    fi
+  fi
+}
+
+# ======================================================================
+# Existing install handling code
+
+update() {
+  updater="${ndprefix}/usr/libexec/netdata/netdata-updater.sh"
+
+  if [ -x "${updater}" ]; then
+    if run ${ROOTCMD} "${updater}" --not-running-from-cron; then
+      progress "Updated existing install at ${ndprefix}"
+      return 0
+    else
+      fatal "Failed to update existing Netdata install at ${ndprefix}"
+    fi
+  else
+    return 1
+  fi
+}
+
+check_for_existing_install() {
+  if pkg_installed netdata; then
+    ndprefix="/"
+  else
+    ndpath="$(command -v netdata 2>/dev/null)"
+    if [ -z "$ndpath" ] && [ -x /opt/netdata/bin/netdata ]; then
+      ndpath="/opt/netdata/bin/netdata"
+    fi
+
+    if [ -n "${ndprefix}" ]; then
+      ndprefix="$(dirname "$(dirname "${ndpath}")")"
+    fi
+
+    if [ "${ndprefix}" = /usr ]; then
+      ndprefix="/"
+    fi
+  fi
+
+  if [ -n "${ndprefix}" ]; then
+    typefile="${ndprefix}/etc/netdata/.install-type"
+    if [ -r "${typefile}" ]; then
+      # shellcheck disable=SC1090
+      . "${typefile}"
+    else
+      INSTALL_TYPE="unknown"
+    fi
+  else
+    return 1
+  fi
+}
+
+handle_existing_install() {
+  check_for_existing_install
+
+  if [ -n "${ndprefix}" ]; then
+    case "${INSTALL_TYPE}" in
+      kickstart-*|legacy-*|manual-static)
+        if [ -z "${NETDATA_REINSTALL}" ]; then
+          ret=0
+
+          if [ "${NETDATA_CLAIM_ONLY}" -eq 0 ]; then
+            if ! update; then
+              warning "Unable to find usable updater script, not updating existing install at ${ndprefix}."
+            fi
+          fi
+
+          if [ -n "${NETDATA_CLAIM_TOKEN}" ]; then
+            progress "Attempting to claim existing install at ${ndprefix}."
+            claim
+            ret=$?
+          elif [ "${NETDATA_CLAIM_ONLY}" -eq 1 ]; then
+            fatal "User asked to claim, but did not proide a claiming token."
+          fi
+
+          exit $ret
+        else
+          progress "Found an existing netdata install at ${ndprefix}, but user requested reinstall, continuing."
+        fi
+        ;;
+      binpkg-*)
```

Comment:
Hi @Ferroin !

What is the procedure for when the script is re-run (without claim parameters) and detects an existing `binpkg-*` installation? What do we expect? Right now the output is kinda vague, i.e.:

```
 --- Using /tmp/netdata-kickstart-bydKfGbLt0 as a temporary directory. --- 
```

And bails out. Should we print something like, `Already installed, thanks very much` ? Try to find and download newer version?

Thanks!

Reply:
The plan is to eventually support binary packages in the updater script, at which point that code path will be merged with the existing one for static and kickstart installs.